tests/compile-fail/enums_clike.rs

```diff
@@ -0,0 +1,53 @@
+#![feature(plugin, associated_consts)]
+#![plugin(clippy)]
+#![deny(clippy)]
+
+#![allow(unused)]
+
+#[repr(usize)]
+enum NonPortable {
+    X = 0x1_0000_0000, //~ ERROR: Clike enum variant discriminant is not portable to 32-bit targets
+    Y = 0,
+    Z = 0x7FFF_FFFF,
+    A = 0xFFFF_FFFF,
+}
+
+enum NonPortableNoHint {
+    X = 0x1_0000_0000, //~ ERROR: Clike enum variant discriminant is not portable to 32-bit targets
```

Comment:
related issues https://github.com/rust-lang/rust/issues/24290 and UB because of it https://github.com/rust-lang/rust/issues/31886